binary_transparency/firmware/cmd/ft_personality/internal/http/server.go

```diff
@@ -339,6 +339,42 @@ func (s *Server) getFirmwareImage(w http.ResponseWriter, r *http.Request) {
 	w.Write(image)
 }
 
+// addAnnotationMalware handles requests to annotate a logged firmware with a malware annotation.
+func (s *Server) addAnnotationMalware(w http.ResponseWriter, r *http.Request) {
+	ss := api.SignedStatement{}
+
+	// Store the original bytes as statement to avoid a round-trip (de)serialization.
+	rawStmt, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	if err := json.NewDecoder(bytes.NewReader(rawStmt)).Decode(&ss); err != nil {
+		http.Error(w, fmt.Sprintf("failed to decode statement: %q", err.Error()), http.StatusBadRequest)
+		return
+	}
+
+	// TODO(mhutchinson): This only works if the signer is the firmware publisher. Support more public keys.
```

Comment:
This is the other discussion point. I propose having N sets of public keys; one for each of the types (firmware, malware annotation, build annotation, ...). This is a crude but effective mechanism for access control.

Reply:
Agreed that we will have a key-pair for each "type" of statement.

Reply:
This is now done.